api/router.proto

```diff
@@ -24,49 +23,64 @@ option go_package = "go.thethings.network/lorawan-stack-legacy/pkg/ttnpb";
 message UplinkMessage {
   bytes payload = 1;
   // ProtocolMessage message = 2;
-  ProtocolRxMetadata protocol_metadata = 11 [(gogoproto.nullable) = false];
-  GatewayRxMetadata gateway_metadata = 12 [(gogoproto.nullable) = false];
+  ProtocolRxMetadata protocol_metadata = 11;
+  GatewayRxMetadata gateway_metadata = 12;
   Trace trace = 21;
 }
 
 message DownlinkMessage {
   bytes payload = 1;
   // ProtocolMessage message = 2;
-  ProtocolTxConfiguration protocol_configuration = 11 [(gogoproto.nullable) = false];
-  GatewayTxConfiguration gateway_configuration = 12 [(gogoproto.nullable) = false];
+  ProtocolTxConfiguration protocol_configuration = 11;
+  GatewayTxConfiguration gateway_configuration = 12;
   Trace trace = 21;
 }
 
 message Trace {
-  string id = 1 [(gogoproto.customname) = "ID"];
+  string id = 1;
   int64 time = 2;
-  string service_id = 3 [(gogoproto.customname) = "ServiceID"];
+  string service_id = 3;
   string service_name = 4;
   string event = 5;
   map<string, string> metadata = 6;
   repeated Trace parents = 11;
 }
 
 message ProtocolRxMetadata {
-  LoRaWANMetadata lorawan = 1 [(gogoproto.customname) = "LoRaWAN"];
+  LoRaWANMetadata lorawan = 1;
 }
 
 message ProtocolTxConfiguration {
-  LoRaWANTxConfiguration lorawan = 1 [(gogoproto.customname) = "LoRaWAN"];
+  LoRaWANTxConfiguration lorawan = 1;
 }
 
 enum Modulation {
   LORA = 0;
   FSK  = 1;
 }
 
+enum Band {
+  EU_863_870 = 0;
+  US_902_928 = 1;
+  CN_779_787 = 2;
+  EU_433     = 3;
+  AU_915_928 = 4;
+  CN_470_510 = 5;
+  AS_923     = 6;
+  AS_920_923 = 61;
+  AS_923_925 = 62;
```

Comment:
Isn't the band ID`AS_923`?

Reply:
Hmm yeah... @htdvisser how is this being used exactly? Is this rather a frequency plan than a band?

Reply:
It's a frequency plan, see

https://github.com/TheThingsNetwork/api/blob/master/protocol/lorawan/lorawan.proto